dot/network/utils_test.go

```diff
@@ -11,6 +11,27 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+const portsAmount = 7100
+
+type portsQueue chan int
+
+func (p *portsQueue) get() int {
+	return <-*p
+}
+
+func (p *portsQueue) put(port int) {
+	*p <- port
+}
+
+var availablePorts portsQueue
+
+func init() {
+	availablePorts = make(portsQueue, portsAmount)
+	for port := 7001; port <= portsAmount; port++ {
+		availablePorts <- port
+	}
```

Comment:
Interesting way to do it! Although the problem is it will fail if you have anything else listening on port `7001` to `14200`.
Would there be a way to bind on port `0` and let the OS assign a free port?

For example to bind an HTTP server on a random available port and get the port back (for tests and/or logs), you can use:

```go
listener, err := net.Listen("tcp", ":0")
if err != nil {
	// ...
}
address := listener.Addr().String()

server := http.Server{Handler: myHandler}
// ...
err = server.Serve(listener)
```

This should work for all TCP listeners, and there is an equivalent API for UDP if that's what we're using. Maybe another PR to this branch would be interesting :thinking: That might also allow users to set `0` as the port to let the OS pick a random port and we can just log out the listening port.

Reply:
@qdm12 thanks for the suggestion, I changed to use `addr := net.ResolveTCPAddr("tcp", "localhost:0")` and `l := net.ListenTCP("tcp", addr)` then the OS can resolve an available port (`port := l.Addr().(*net.TCPAddr).Port`) for each call!

Reply:
Continued in https://github.com/ChainSafe/gossamer/pull/2105#discussion_r764713206